bindings/CXX11/adios2/cxx11/Engine.h

```diff
@@ -143,6 +169,21 @@ class Engine
     void Put(Variable<T> variable, const T *data,
              const Mode launch = Mode::Deferred);
 
+    /**
+     * Put data associated with a container-like thing, e.g., a Kokkos::View
+     */
+    template <
+        class T, class C,
+        typename = typename std::enable_if<
+            detail::is_container<C>::value &&
+            std::is_same<typename std::remove_cv<typename C::value_type>::type,
```

Comment:
std::is_same<typename std::remove_cv<typename C::value_type>::type,
            
            how about moving this to the is_container trait?

Reply:
Actually, this makes sure that the `value_type` of the container is consistent with the adios2 `Variable`, so I can't move it into `is_container`, since it's not a property of the container.

I did change this into a `static_assert`, though, so instead of a missing overload error one gets a somewhat more specific error message when trying to, e.g., write a `Kokkos::View<float*>` into a `Variable<double>`.

Reply:
FWIW,  I did change this back later to the same behavior as the current `Put(Variable<T>, const T*, ...)`, ie., only matching types are considered an overload (kinda like `Put(Variable<T>, const Kokkos::View<T*>, ..)`, so if the View's value_type doesn't match the variable it's just not considered in overload resolution)